chain/chain/src/state_sync.rs

```diff
@@ -154,6 +199,34 @@ pub(crate) fn update_sync_hashes<T: ChainStoreAccess>(
         return remove_old_epochs(chain_store.store(), store_update, header, &prev_header);
     }
 
-    save_epoch_new_chunks(chain_store, store_update, header)?;
-    remove_old_blocks(chain_store, store_update, header)
+    on_new_header(chain_store, store_update, header)
+}
+
+///. Returns whether `block_hash` is the block that will appear immediately before the "sync_hash" block. That is,
+/// whether it is going to be the prev_hash of the "sync_hash" block, when it is found.
+///
+/// `block_hash` is the prev_hash of the future "sync_hash" block iff the number of new chunks in the epoch in
+/// each shard is at least 2, and at least one shard of the block `prev_hash` doesn't yet have 2 new chunks in it.
```

Comment:
nit: Maybe say "is the first block in the that has the X property" instead of "is a block that has the X property but the previous block does not"

Reply:
yea thats better, done